plugins/kubernetes-backend/src/service/KubernetesProxy.ts

```diff
@@ -135,20 +135,26 @@ export class KubernetesProxy {
       const logger = this.logger.child({ cluster: originalCluster.name });
       middleware = createProxyMiddleware({
         logProvider: () => logger,
+        ws: true,
         secure: !originalCluster.skipTLSVerify,
         changeOrigin: true,
+        pathRewrite: { [`^${originalReq.baseUrl}`]: '' },
         router: async req => {
-          // Re-evaluate the cluster on each request, in case it has changed
-          const cluster = await this.getClusterForRequest(req);
+          const isWS = req.headers.upgrade === 'websocket';
+          const cluster = isWS
+            ? originalCluster
+            : await this.getClusterForRequest(req);
           const url = new URL(cluster.url);
+          const ca = bufferFromFileOrString('', cluster.caData)?.toString();
+
+          // Re-evaluate the cluster on each request, in case it has changed
```

Comment:
I think this comment should stay above the line where `getClusterForRequest` is called (IMO the purpose of the comment is to justify that call), or else this comment should just be deleted

Reply:
Was in a bit of a rush yesterday night, will move it.

Reply:
Reverted some changes to try a slightly different approach.